src/WebJobs.Script/ScriptHostBuilderExtensions.cs

```diff
@@ -411,6 +411,26 @@ internal static void ConfigureApplicationInsights(HostBuilderContext context, IL
                         o.EnableDependencyTracking = false;
                     });
                 }
+
+                builder.Services.AddOptions<LoggerFilterOptions>().Configure<IEnvironment>((options, environment) =>
+                {
+                    // Skip sending user generated logs to AI and QuickPulse if worker AI agent is configured, worker will send these logs to AI and Quickpulse service.
+                    if (environment.IsApplicationInsightsAgentEnabled())
+                    {
+                        options.AddFilter<ApplicationInsightsLoggerProvider>((category, logLevel) =>
+                        {
+                            // match Function.<FunctionName>.User
+                            if (category.Length > 14 && category.EndsWith(".User", StringComparison.Ordinal))
```

Comment:
Is `category` guaranteed to be non-null here?

Reply:
yes, this is what my understanding is. I will double check with Brett otherwise I will add a null check to be safe.

Reply:
Added null check.